app/modules/asset_groups/schemas.py

```diff
@@ -42,6 +53,47 @@ class Meta(BaseAssetGroupSightingSchema.Meta):
         dump_only = BaseAssetGroupSightingSchema.Meta.dump_only
 
 
+class AssetGroupSightingAsSightingSchema(BaseAssetGroupSightingSchema):
+    """
+    In order for the frontend to render an AGS with the same code that renders a
+    sighting, we have to pop out all the fields in AGS.config into the top-level
+    of the schema. This is done with AssetGroupSighting.config_field_getter,
+    which creates a getter for a config field of a given name. We can add more
+    fields using the pattern below.
+    """
+
+    assets = base_fields.Function(AssetGroupSighting.get_assets)
+    completion = base_fields.Function(AssetGroupSighting.get_completion)
+
+    # Note: these config_field_getter vars should conform to SIGHTING_FIELDS_IN_AGS_CONFIG
+    # at the top of this file
+    decimalLatitude = base_fields.Function(
```

Comment:
> I don't see a way to flexibly add fields to the schema besides hard-coding them. This means we return None/null field values rather than not returning those fields, on e.g. decimalLatitude

The only way I found is by adding a post processor in `AssetGroupSightingAsSightingSchema`:

```python
from marshmallow import post_dump
...
class AssetGroupSightingAsSightingSchema(BaseAssetGroupSightingSchema):
    ...
    @post_dump
    def remove_none_fields(self, data):
        for field in ('decimalLatitude', 'decimalLongitude'):
            if data[field] is None:
                del data[field]
        return data
```

(I found this by looking into the marshmallow 2.21.0 code)

Reply:
Now here's a place where we should use SIGHTING_FIELDS_IN_AGS_CONFIG within the schema file, in place of decimalLatitude, decimalLongitude in your example. I have done this now.